### ISSUE TITLE
Managing external links to internal parts of the app when we have app.posthog.com and eu.posthog.com

### ISSUE THREAD

**Issue by lharries** [PostHog/posthog#15114]
(I'm happy to build this but would be good to get others input on it)

TL;DR if you go to a url such as https://app.posthog.com/login?next=/dashboard_templates/user_research when you are only logged in on eu.posthog.com not app.posthog then show a message and autoredirect after a few seconds

## Is your feature request related to a problem?

Problem:
- We often want to link to areas of the app (e.g. the settings page or specific dashboard templates) from our own website.
- We also want others users to be easily able to share links to internal parts of our app (e.g. when supporting each other via the community, sharing links on twitter to new features)
- However, we have two subdomains - eu.posthog.com and app.posthog.com and currently the links don't work across them

## Describe the solution you'd like

- When a user tries to visit a link on one of the subdomains which isn't `/`, `/signup` or `/login`, AND they are logged in to the other subdomain, then shows a redirecting message and redirect them after a few seconds
  - Why this? It means us and customers can both share any link and it should just work regardless if it's eu or app.posthog.com
  - We could also add a query parameter which skips the redirect message e.g. if linking from posthog.com and we know we want to go to the one they are logged in on

- What if users have an app.posthog.com and an eu.posthog.com login?
  - By default, do nothing and just take them to the link they clicked on. As most of these links will probably be internal links shared to the correct domain. If they need to, they can change the url directly as most of our ICP customers are savvy enough to realize and do this.
  - In in the unlikely case there's lots of confusion around people being logged into both and they are clicking on links on our marketing site, we could either add a query parameter to explicitly show a redirect message OR have a dedicated subdomain e.g. redirect.posthog.com However, this would only work when people are willing to manually change the links (most likely when we add links to our posthog.com site)

## Describe alternatives you've considered

- the url already automatically translates https://eu.posthog.com/feature_flags/5678 to https://eu.posthog.com/login?next=/feature_flags/5678 if they are not logged in. And when they change the data region from EU to US it automatically takes them to the link. So we could just do nothing however I think it's a better experience
- redirect.posthog.com or posthog.com/redirect However, this would only work when people are willing to manually change the links

**Comment by lharries** [comment]
@andyvan-ph / @ivanagas / @joethreepwood / @PostHog/website-docs this would mean you can add internal links to your docs which go to app.posthog.com and if they are instead logged into eu.posthog.com it will redirect them to the right one

Let me know if you have a better idea or a way you are working around this already

**Comment by joethreepwood** [comment]
This definitely seems like the best solution, to me. 

This could pretty massively improve the experience with tutorials and onboarding emails, etc., so would be a huge benefit to Marketing. Currently we default to linking to a `/login` page, or content. Pushing users into the app would be better for them, and us.

**Comment by joethreepwood** [comment]
Just adding a note, as discussed, that if we can find a way to include the 'Get support' modal into this as well then that would be 🔥

**Comment by raquelmsmith** [comment]
> When a user tries to visit a link on one of the subdomains which isn't /, /signup or /login, AND they are logged in to the other subdomain, then shows a redirecting message and redirect them after a few seconds

Why not redirect on these paths? It'd be nice to have this work for /login from the marketing website so people are taken to the correct instance.

**Comment by lharries** [comment]
> Why not redirect on these paths? It'd be nice to have this work for /login from the marketing website so people are taken to the correct instance.

Good point

Currently the logic is implemented in the login page, so that works. However for the sign up page would need a bit of extra work plus I think it's more reasonable that someone might be logged into one instance and actively sign up for another instance. The data region selector is also very obvious for them

PR up for this here: https://github.com/PostHog/posthog/pull/15119
